roles/edpm_nova/scripts/derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,587 @@
+                        derivations of other configurations, the derived configuration shall
+                        replace the original user_config.
+                        """
+                        del_user_config = True
+
+    if sel_addr:
+        pci_passthrough = get_pci_passthrough_whitelist(
+            user_config, pf_name, sel_addr)
+    return pci_passthrough, del_user_config
+
+
+def get_passthrough_config_for_all_pf(user_config,
+                                  system_configs,
+                                  allocated_pci):
+    derived_config = []
+    nic_partition_pfs = get_sriov_nic_partition_pfs(system_configs)
+    non_nic_partition_pfs = get_sriov_non_nic_partition_pfs(system_configs)
+    del_user_config = False
+
+    """
+    For each user_config, do the matching for NIC Partitioning PFs/VFs
+    """
+    for pf in nic_partition_pfs:
+        passthrough_tmp, status = get_passthrough_config(
+            user_config, pf, allocated_pci, False)
+        del_user_config = del_user_config or status
+        if passthrough_tmp:
+            derived_config.extend(passthrough_tmp)
+
+    """
+    If there is no config added from NIC Part nics, then skip parsing the
+    NON NIC Partitioned PFs.
+    """
+    if (derived_config or del_user_config):
+        for pf in non_nic_partition_pfs:
+            passthrough_tmp, status = get_passthrough_config(
+                user_config, pf, allocated_pci, True)
+            derived_config.extend(passthrough_tmp)
+
+    if derived_config:
+        return derived_config, False
+    else:
+        return derived_config, del_user_config
+
+
+# Todo: (remove)This function will not be needed now
+def get_neutron_devmap_configs():
+    ml2_sriov = []
+    valid_spec = 'physical_device_mappings'
+
+    #Read phydev mapping from neutron conf file
+    try:
+        with open('/var/lib/os-net-config/01-neutron-sriov-agent.conf') as neutron_config:
+            for line in neutron_config:
+                line = line.replace('\n', '')
+                if line.startswith(valid_spec):
+                    phy_map = re.split(', |,|=|= ', line)
+                    for i in phy_map:
+                        if ':' in i:
+                            ml2_sriov.append(i)
+    except IOError:
+        ml2_sriov = []
+    return  ml2_sriov
+
+
+def get_pf_name_from_phy_network(physical_network):
+    out = get_neutron_devmap_configs()
+    if len(out):
+        phys_dev_mappings = json.loads(out)
+        ''' Check the data type of first json decode '''
+        if not isinstance(phys_dev_mappings, list):
+            msg = f'ml2::sriov::physical_device_mappings specified is not a list {phys_dev_mappings}'
+            raise InvalidConfigException(msg)
+
+        for phy_dev_map in phys_dev_mappings:
+            net_name, nic_name = phy_dev_map.split(':')
+            if net_name == physical_network:
+                return nic_name
+        return None
```

Comment:
@vcandapp Need to change this logic based on edpm_neutron_sriov_agent_SRIOV_NIC_physical_device_mappings edpm-ansible variable.
you can default nova variable with getting value from above variable and use added variable for this logic.

Reply:
Updated default.yml with sriov conf file name as user configurable input